### ISSUE TITLE
Cookie auth broken when 'authentication_handlers' added to local.ini

### ISSUE THREAD

**Issue by cgwic** [apache/couchdb#1311]
## Expected Behavior
```
curl -vX GET 192.168.99.100:5984/_session --cookie AuthSession=ZGlqbzo...
{"ok":true,"userCtx":{"name":"name","roles":["role"]},"info":{"authentication_db":"_users","authentication_handlers":["cookie","default"],"authenticated":"cookie"}}
```

## Current Behavior
```
curl -vX GET 192.168.99.100:5984/_session --cookie AuthSession=ZGlqbzo...
{"ok":true,"userCtx":{"name":null,"roles":[]},"info":{"authentication_db":"_users","authentication_handlers":["cookie","default"]}}
```

## Possible Solution
comment this string in [chttpd] section in local.ini file
`authentication_handlers = {couch_httpd_auth, cookie_authentication_handler}, {couch_httpd_auth, default_authentication_handler}`

## Steps to Reproduce (for bugs)
1. Add `authentication_handlers = {couch_httpd_auth, cookie_authentication_handler}, {couch_httpd_auth, default_authentication_handler}` in [chttpd] section in local.ini file
2. Restart couchdb
3. Try to get access with nonadmin user using cookie auth

## Context
It's not a bug for me just unexpected behaviour. But i'm new to CouchDB and it takes some time to find cause of this.

## Your Environment
* Version used: CouchDB 2.1.1
* Browser Name and version: curl 7.58.0, chrome 66.0.3359
* Operating System and version (desktop or mobile): Windows 7, Android 5.1, Linux default 4.9.93-boot2docker
* Link to your project: private

**Comment by cgwic** [comment]
@wohali yes, now i know, but anyway, why auth get broken when the same thing in config?

**Comment by wohali** [comment]
Because you put `couch_httpd_auth` in the config for `chttpd`, when that handler is only written to work with the original `couch_httpd` module. the `chttpd_auth` module is written for `chttpd`, so you have to use that version instead.

It would be like using a 64-bit library with a 32-bit application - it's simply incompatible.

**Comment by cgwic** [comment]
@wohali O, I see, thank you. Really didn't pay attention for prefix.
